Automattic/pocket-casts-ios
# Update account upgrade modal

Fixes #

This PR updates the account upgrade model to match the Android designs.

Instead of sending users to the Onboarding upgrade flow design we restore the modal to select a period and then allow purchase.

| 50% offer annual | no offer monthly | 1 Month Free annual | no offer monthly }
 | - | - | - | - | 
| ![Simulator Screenshot - iPhone 15 Pro - 2024-02-06 at 19 35 37](https://github.com/Automattic/pocket-casts-ios/assets/651601/49161742-cd1d-4692-bcf6-bb39bb3ac4c2) | ![Simulator Screenshot - iPhone 15 Pro - 2024-02-06 at 19 35 39](https://github.com/Automattic/pocket-casts-ios/assets/651601/2a23de5a-a310-471d-9a93-a5b312830222) | ![Simulator Screenshot - iPhone 15 Pro - 2024-02-06 at 19 35 47](https://github.com/Automattic/pocket-casts-ios/assets/651601/340abf6e-cb10-4e48-a9da-09402e4140df) | ![Simulator Screenshot - iPhone 15 Pro - 2024-02-06 at 19 35 49](https://github.com/Automattic/pocket-casts-ios/assets/651601/77d316da-97c3-4c16-8b48-a3ca4506cbe2) | 


## To test

 - Edit the podcast scheme to use the "Pocket Casts Configuratio.storekit" file for the StoreKitConfiguration
 - Start the app on the simulator
 - Go to the Profile tap ( you must have an account with no active subscription)
 - Tap on Account
 - Check if the cards show correctly for the upgrade options
 - Tap on Subscribe to Plus
 - Check if a modal is show and match the design above ( on smaller devices it should show a taller modal)
 - Switch between the Yearly and Monthly options and see if it shows correctly
 - Tap on Subscribe to Plus
 - Check if the IAP flow starts correctly 

## Checklist
